client/extensions/woocommerce/app/promotions/promotion-form-conditions-card.js

```diff
@@ -0,0 +1,268 @@
+		const onSelectEndDay = ( day ) => onChange( { target: { name: 'date_on_sale_to_gmt', value: day } } );
+
+		return (
+			<div className="promotions__promotion-form-condition-daterange">
+				<DatePicker
+					initialMonth={ startDate }
+					selectedDay={ startDate }
+					onSelectDay={ onSelectStartDay }
+				/>
+				<DatePicker
+					initialMonth={ endDate }
+					selectedDay={ endDate }
+					onSelectDay={ onSelectEndDay }
+				/>
+			</div>
+		);
+	},
+};
+
+const otherCoupons = {
+	name: 'otherCoupons',
+	getText: ( translate ) => translate( 'Can be applied in conjunction with other coupons' ),
+	isChecked: ( promotion ) => {
+		const { individual_use } = promotion.coupon || {};
+		return individual_use || false;
+	},
+	setChecked: ( checked, promotion, siteId, editPromotion ) => {
+		const individual_use = checked;
+		editPromotionCoupon( siteId, promotion, { individual_use }, editPromotion );
+	},
+	renderEdit: noop,
+};
+
+const allowWithSale = {
+	name: 'allowWithSale',
+	getText: ( translate ) => translate( 'Can be applied to products on sale' ),
+	isChecked: ( promotion ) => {
+		const { exclude_sale_items } = promotion.coupon || {};
+		if ( 'undefined' === typeof exclude_sale_items ) {
+			// If it wasn't set before, default to false.
+			return false;
+		}
+		return ! exclude_sale_items;
+	},
+	setChecked: ( checked, promotion, siteId, editPromotion ) => {
+		const exclude_sale_items = ! checked;
+		editPromotionCoupon( siteId, promotion, { exclude_sale_items }, editPromotion );
+	},
+	renderEdit: noop,
+};
+
+const changeValues = {
+	minimum_amount: ( value, promotion, siteId, editPromotion ) => {
+		editPromotionCoupon( siteId, promotion, { minimum_amount: value }, editPromotion );
+	},
+	maximum_amount: ( value, promotion, siteId, editPromotion ) => {
+		editPromotionCoupon( siteId, promotion, { maximum_amount: value }, editPromotion );
+	},
+	date_expires_gmt: ( value, promotion, siteId, editPromotion ) => {
+		editPromotionCoupon( siteId, promotion, { date_expires_gmt: value }, editPromotion );
+	},
+	date_on_sale_from_gmt: ( value, promotion, siteId, editPromotion ) => {
+		editPromotionProduct( siteId, promotion, { date_on_sale_from_gmt: value }, editPromotion );
+	},
+	date_on_sale_to_gmt: ( value, promotion, siteId, editPromotion ) => {
+		editPromotionProduct( siteId, promotion, { date_on_sale_to_gmt: value }, editPromotion );
+	},
+};
+
+const couponConditions = {
+	minMaxSpend,
+	expiryDate,
+	otherCoupons,
+	allowWithSale,
+};
+
+const productSaleConditions = {
+	dateRange,
+};
+
+function renderCondition( promotion, condition, currency, translate, onCheck, onChange ) {
+	return (
+		<FormFieldset key={ condition.name } className="promotions__promotion-form-condition">
+			<FormInputCheckbox checked={ condition.isChecked( promotion ) } value={ condition.name } onChange={ onCheck } />
+			<FormLabel>{ condition.getText( translate ) }</FormLabel>
```

Comment:
I _think_ if you make FormInputCheckbox a child of FormLabel, that will let you click on the label to check and uncheck the box - much better for accessibility.

e.g. see https://github.com/Automattic/wp-calypso/blob/b9f6783bbaf733790c3d6b4e0a7e27f241d6b64e/client/extensions/woocommerce/app/order/order-fulfillment/index.js#L164

Reply:
Thanks, I've started doing this with all checkboxes.